File: csharp/2016/blood-father-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Blood Father" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Blood Father",
                year: 2016,
                plot: "An ex-con reunites with his estranged wayward 17-year old daughter to protect her from drug dealers who are trying to kill her.",
                rating: 6.4
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Blood Father",
                year: 2016
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2016/blood-father-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Blood Father to the database
movies.insert(
    title="Blood Father", 
    year=2016, 
    plot="An ex-con reunites with his estranged wayward 17-year old daughter to protect her from drug dealers who are trying to kill her.", 
    rating=6.4
)

# Confirm that the movie was added
movie = movies.select(
    title="Blood Father", 
    year=2016
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    